### PR TITLE
[IBM WebSphere Application Server] Add WebSphere Application Server Integration Package with Servlet data stream 

### PR DESCRIPTION
**Pull request by kush-elastic**
- Enhancement

## What does this PR do?

- Generated the skeleton of WebSphere Application Server integration package.
- Added 1 data stream (Servlet).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added pipeline test cases for the data stream.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/websphere_application_server directory.
- Run the following command to run tests.
`elastic-package test`

## Related issues

- Relates #2847
- Relates #2922
- Relates #3138
- Relates #3199

## Screenshots

websphere_application_server-servlet-dashboard-screenshot

### CODE REVIEW

File: packages/websphere_application_server/data_stream/servlet/_dev/test/pipeline/test-servlet-metrics.json-expected.json [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+                "prometheus"
+            ],
+            "websphere_application_server": {
+                "servlet": {
+                    "app_name": "isclite#isclite.war",
+                    "async_context": {
+                        "response_time_seconds": 0,
+                        "total_responses": 0
+                    },
+                    "concurrent_requests": 0,
```
Comment (yug-elastic) [inline packages/websphere_application_server/data_stream/servlet/_dev/test/pipeline/test-servlet-metrics.json-expected.json:26]: How about grouping `concurrent_requests` and `requests_processed` to `requests.concurrent` and `requests.processed`?

File: packages/websphere_application_server/data_stream/servlet/_dev/test/pipeline/test-servlet-metrics.json-expected.json [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+                    "errors": 0,
+                    "requests_processed": 0,
+                    "response_time_seconds": 0,
+                    "responses_processed": 0
```
Comment (yug-elastic) [inline packages/websphere_application_server/data_stream/servlet/_dev/test/pipeline/test-servlet-metrics.json-expected.json:30]: If we change `requests_processed` to `requests.processed`, I think we should also nest this like `responses.processed`.

File: packages/websphere_application_server/data_stream/servlet/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      description: Number of errors that were generated while responding to a request.
+    - name: concurrent_requests
+      type: long
+      description: Number of concurrent requests sent to the servlet.
+    - name: uri
+      type: group
+      fields:
+        - name: total_requests
+          description: >-
+            total number of requests that a servlet processed for the specified URL.
```
Comment (yug-elastic) [inline packages/websphere_application_server/data_stream/servlet/fields/fields.yml:33]: nit:
```suggestion
            Total number of requests that a servlet processed for the specified URL.
```

File: packages/websphere_application_server/data_stream/servlet/fields/fields.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          type: double
+        - name: total_responses
+          description: The total number of responses for the specified URL.
+          type: long
+        - name: async_context
+          type: group
+          fields:
+            - name: response_time_seconds
+              description: >-
+                The total time spent(in seconds) per URL for the AsyncContext response to complete.
```
Comment (yug-elastic) [inline packages/websphere_application_server/data_stream/servlet/fields/fields.yml:51]: nit:
```suggestion
                The total time spent (in seconds) per URL for the AsyncContext response to complete.
```

File: packages/websphere_application_server/data_stream/servlet/fields/fields.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            - name: total_responses
+              description: >-
+                The total number of AsyncContext responses for the specified URL.
+              type: long
+    - name: async_context
+      type: group
+      fields:
+        - name: response_time_seconds
+          description: >-
+            The total time spent(in seconds) per servlet for the AsyncContext response to complete.
```
Comment (yug-elastic) [inline packages/websphere_application_server/data_stream/servlet/fields/fields.yml:62]: nit:
```suggestion
            The total time spent (in seconds) per servlet for the AsyncContext response to complete.
```

**Review by yug-elastic — CHANGES_REQUESTED** [review]
Requested some changes.

File: packages/websphere_application_server/data_stream/servlet/_dev/test/pipeline/test-servlet-metrics.json-expected.json (hunk 1 above)
Comment (kush-elastic) [reply packages/websphere_application_server/data_stream/servlet/_dev/test/pipeline/test-servlet-metrics.json-expected.json:26]: Makes sense, thanks!

**Review by yug-elastic — APPROVED** [review]
LGTM!

**Review by ishleenk17 — COMMENTED** [review]
How about the docker compose file in this data stream ?

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**